.github/workflows/test-e2e.yaml

```diff
@@ -0,0 +1,69 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# The test-e2e action does not check out the repo. Instead, it triggers the e2e
+# Cloud Build using this action's commit SHA, then waits for the cloud build
+# run to complete.
+#
+# This action uses a Cloud Build trigger rather than `gcloud build submit`
+# because we want Cloud Build to check out the repo. We don't want to upload
+# the source from this github action.
+#
+# We don't want e2e build log to be printed in this Github Action's log
+# because it may contain secrets.
+#
+name: e2e-test-trigger
+on:
+  pull_request:
+  pull_request_target:
+    types: [labeled]
+jobs:
+  build:
+    if: "${{ github.event.action != 'labeled' || github.event.label.name == 'e2e: run' }}"
+    name: Trigger E2E test build in Google Cloud Build
+    runs-on: ubuntu-latest
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    steps:
+      - name: Remove PR Label
+        if: "${{ github.event.action == 'labeled' && github.event.label.name == 'e2e: run' }}"
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            try {
+              await github.rest.issues.removeLabel({
+                name: 'e2e: run',
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                issue_number: context.payload.pull_request.number
+              });
+            } catch (e) {
+              console.log('Failed to remove label. Another job may have already removed it!');
+            }
+      - id: 'auth'
+        name: 'Authenticate to Google Cloud'
+        uses: 'google-github-actions/auth@v1.0.0'
+        with:
+          workload_identity_provider: ${{ secrets.PROVIDER_NAME }}
+          service_account: ${{ secrets.SERVICE_ACCOUNT }}
+          access_token_lifetime: 600s
+          project_id: ${{ secrets.GOOGLE_CLOUD_PROJECT }}
+          create_credentials_file: true
+      - name: 'Set up Cloud SDK'
+        uses: 'google-github-actions/setup-gcloud@v1'
+      - id: 'trigger'
+        name: 'Start E2E test using Cloud Build'
+        run: 'gcloud builds triggers run e2e-test-pr --sha "${{ github.event.pull_request.head.sha }}" --billing-project "${{secrets.GOOGLE_CLOUD_PROJECT}}" --project="${{secrets.GOOGLE_CLOUD_PROJECT}}" --format="table(metadata.build.id,metadata.build.logUrl)"'
```

Comment:
After considering, I think I want to continue to use `gcloud build trigger` instead of `gcloud build submit`. The main reason is that our docker container images need to be built in a way that captures the provenance. By building with a Cloud Build trigger, Cloud Build will annotate the resulting docker container with the appropriate git repo information.